clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/security/AmazonBasicCredentialsLoader.java

```diff
@@ -101,4 +111,109 @@ public void load() {
+      log.info(
+          "Multi-threading is enabled for loading aws accounts. Using {} threads, with timeout: {}s",
+          credentialsConfig.getLoadAccounts().getNumberOfThreads(),
+          credentialsConfig.getLoadAccounts().getTimeoutInSeconds());
+      toApply = multiThreadedParseAccounts(definitions);
+    } else {
+      log.info("Multi-threading is disabled. AWS accounts will be loaded serially");
+      for (T definition : definitions) {
+        if (!loadedDefinitions.containsKey(definition.getName())) {
+          U cred = parser.parse(definition);
+          if (cred != null) {
+            toApply.add(cred);
+            // Add to loaded definition now in case we trigger another parse before this one
+            // finishes
+            loadedDefinitions.put(definition.getName(), definition);
+          }
+        } else if (!loadedDefinitions.get(definition.getName()).equals(definition)) {
+          U cred = parser.parse(definition);
+          if (cred != null) {
+            toApply.add(cred);
+            loadedDefinitions.put(definition.getName(), definition);
+          }
+        }
+      }
+    }
+
+    log.info("saving aws accounts in the credentials repository");
+    Stream<U> stream = parallel ? toApply.parallelStream() : toApply.stream();
+    stream.forEach(credentialsRepository::save);
+    log.info("parsed and saved {} aws accounts", credentialsRepository.getAll().size());
+  }
+
+  /**
+   * parses aws accounts using a configurable fixed thread pool.
+   *
+   * @param definitions - the list of aws accounts to parse
+   * @return - a list of parsed aws accounts
+   */
+  private List<U> multiThreadedParseAccounts(Collection<T> definitions) {
+    List<U> toApply = new ArrayList<>();
+    final ExecutorService executorService =
+        Executors.newFixedThreadPool(
+            credentialsConfig.getLoadAccounts().getNumberOfThreads(),
+            new ThreadFactoryBuilder()
+                .setNameFormat(AmazonCredentialsParser.class.getSimpleName() + "-%d")
+                .build());
+
+    final ArrayList<Future<U>> futures = new ArrayList<>(definitions.size());
+    for (T definition : definitions) {
+      if (!loadedDefinitions.containsKey(definition.getName())
+          || !loadedDefinitions.get(definition.getName()).equals(definition)) {
+        futures.add(executorService.submit(() -> parser.parse(definition)));
+      }
+    }
+    for (Future<U> future : futures) {
+      try {
+        U cred =
+            future.get(credentialsConfig.getLoadAccounts().getTimeoutInSeconds(), TimeUnit.SECONDS);
+        if (cred != null) {
+          toApply.add(cred);
+          // Add to loaded definition now in case we trigger another parse before this one finishes
+          definitions.stream()
+              .filter(t -> t.getName().equals(cred.getName()))
+              .findFirst()
+              .ifPresent(definition -> loadedDefinitions.put(cred.getName(), definition));
```

Comment:
Is there ever a case where findFirst wouldn't find a definition here?

Reply:
I don't think so

Reply:
Part of me wants some logging or metric if this fails, but I'm not sure it's worth it.  If it's easy though...

Reply:
done. please see https://github.com/spinnaker/clouddriver/pull/5539/commits/fe475374d8cb4552652fb3326d4a11ee6eeca11b